sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreSuite.scala

```diff
@@ -235,6 +236,162 @@ class StateStoreSuite extends StateStoreSuiteBase[HDFSBackedStateStoreProvider]
     assert(getSizeOfStateForCurrentVersion(store.metrics) > noDataMemoryUsed)
   }
 
+  test("maintenance") {
```

Comment:
Just a head-up: "maintenance" and "snapshotting" are moved from StateStoreSuiteBase to StateStoreSuite as these tests cannot be applied to RocksDB state store provider.

Reply:
Yes, since these tests have the assumption that the provider have a `baseDir`